localstack/services/firehose/provider.py

```diff
@@ -779,3 +795,76 @@ def _get_s3_object_path(self, stream_name, prefix):
         path = pattern.format(pre=prefix, name=stream_name, rand=str(uuid.uuid4()))
         path = timestamp(format=path)
         return path
+
+    def _put_to_redshift(
+        self,
+        stream_name: str,
+        records: List[Dict],
+        redshift_destination_description: RedshiftDestinationDescription,
+    ):
+        jdbcurl = redshift_destination_description.get("ClusterJDBCURL")
+        cluster_id = self._get_cluster_id_from_jdbc_url(jdbcurl)
+        db_name = jdbcurl.split("/")[-1]
+        table_name = redshift_destination_description.get("CopyCommand").get("DataTableName")
+
+        columns_str = None
+        rows_to_insert = []
+        for record in records:
+            data = base64.b64decode(record.get("Data") or record.get("data"))
+            data = to_str(data)
+            data = json.loads(data)
+
+            if columns_str is None:
+                columns_str = ", ".join(data.keys())
+
+            values = []
+            for value in data.values():
+                if isinstance(value, str):
+                    # Escaping single quotes by doubling them
+                    value = value.replace("\t", " ")
+                    value = f"'{value}'"
+                elif value is None:
+                    value = "NULL"
+                else:
+                    value = str(value)
+                values.append(value)
+
+            value_str = ",".join(values)
+            rows_to_insert.append(value_str)
```

Comment:
Suggestion: this feels like it is a concept that could be named. Perhaps extract into a helper function? Then:

```suggestion
        columns_str = None
        rows_to_insert = [_transform_record_for_redshift(record) for record in records]
```

Reply:
extracted into separate functions in: https://github.com/localstack/localstack/commit/051c6630851564b916b556a226a043cfe125b20f